### PR TITLE
🌱 e2e: Enhanced Serial Log Checking Functionality 

### LINKED ISSUE #1466: e2e: Enhanced Serial Log Checking Functionality
**User Story**

As a developer, I need the ability to check structured serial logs for easier debugging and system monitoring. This is crucial for reviewing system actions (like those by IPA) and swiftly pinpointing errors during key processes such as provisioning, enhancing our understanding and troubleshooting efficiency.

**Detailed Description**
I propose a feature that enables the system to automatically save serial logs to the '_artifacts' folder. This will allow developers to review the logs as needed.

### PR DESCRIPTION
**Pull request by maxrantil**
Fixes #1466

### CODE REVIEW

**Review by smoshiur1237 — COMMENTED** [review]
LGTM

**Comment by maxrantil** [comment]
What do you think @lentzi90 ?
/hold

File: hack/ci-e2e.sh [hunk 1]
```diff
@@ -47,6 +47,8 @@ minikube image load quay.io/metal3-io/baremetal-operator:e2e
 # Create libvirt domain
 VM_NAME="bmo-e2e-0"
 export BOOT_MAC_ADDRESS="00:60:2f:31:81:01"
+SERIAL_LOG_DIR="${REPO_ROOT}/test/e2e/_artifacts/logs/baremetal-operator-system/qemu/serial"
```
Comment (lentzi90) [inline hack/ci-e2e.sh:50]: I would interpret `baremetal-operator-system` as the namespace here and it is a bit weird to have the serial logs under this folder. Perhaps just put it directly under logs?
```suggestion
SERIAL_LOG_DIR="${REPO_ROOT}/test/e2e/_artifacts/logs/qemu/serial"
```

**Comment by maxrantil** [comment]
The serial logs disappeared after they were written, so I needed to copy them over to the artifacts folder to save them. How do you like the solution now, @lentzi90 ?

**Review by lentzi90 — COMMENTED** [review]
At what point do they disappear? Can we copy the logs after the test but before the `make clean-e2e`? I would like to keep this outside the go code if possible.

The security issue is no big deal. It is just that we do not know what file we read since it comes from a variable. In test code this is no issue.

**Comment by maxrantil** [comment]
The logs disappear when the namespace is being deleted. That happens just after the LIVE-ISO test so I can't see a way of doing it outside the Go code if we don't change the cleanup code then.

**Comment by lentzi90** [comment]
Can you check how this works in CAPM3? I don't think we touch the serial logs until after all the tests are done there.

**Comment by maxrantil** [comment]
@lentzi90 Is this what you had in mind? The serial logs are now appended after each other in the file, is this the way?

**Review by lentzi90 — COMMENTED** [review]
Excellent!
/lgtm
/cc @kashifest

**Comment by tuminoid** [comment]
Looks clean, nice.

/test-centos-e2e-integration-main
/test-ubuntu-integration-main

**Pull request merged**